tiledb/sm/buffer/buffer.h

```diff
@@ -276,6 +303,148 @@ class Buffer {
   Status ensure_alloced_size(uint64_t nbytes);
 };
 
+// =======================================================
+//      ConstBuffer
+// =======================================================
+/**
+ * A read-only buffer fully-initialized at construction. It does not manage
+ * memory; its storage is subordinate to some other object.
+ */
+class ConstBuffer : public BufferBase {
+ public:
+  /** Default constructor deleted. */
+  ConstBuffer() = delete;
+
+  /**
+   * Ordinary constructor.
+   *
+   * @param data The data of the buffer.
+   * @param size The size of the buffer.
+   */
+  ConstBuffer(const void* data, uint64_t size);
+
+  /**
+   * Constructor from a general-purpose buffer.
+   *
+   * @param buff The buffer the object will encapsulate, working on its
+   *     data and size, but using a separate local offset, without affecting
+   *     the input buffer.
+   */
+  explicit ConstBuffer(Buffer* buff);
+
+  /** Destructor. */
+  ~ConstBuffer() = default;
+
+  /** Returns the buffer data. */
+  const void* data() const;
+
+  /** Returns pointer to data at the current offset. */
+  const void* cur_data() const;
```

Comment:
Should these be in BufferBase ?

Reply:
They can't be because of the const/non-const return value issue.

Reply:
think can be accommodated, as commented above,  
removing 'const' method body modifier, 
tho build would be required to see if that's acceptable to existing callers.